Decision threshold plot

Rick-Mackenbach opened the pull request:
#### Reference Issues/PRs
In reference to this issue: #16153.


#### What does this implement/fix? Explain your changes.
We want to add the first threshold decision to the edges of the decision tree. So for the first split, we add the decision criteria for the left child, and right child.

#### Additional comments
Added a new PR since my previous repository was deleted.

Commit:
"Added changes to newly upstream master"

Commit:
"Nothing"

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
"Cleaned up the code in _export, changed the test files to make them
pass. Not ready yet, need new logic for parsing labels."

Commit:
"Change linting error in test file"

Rick-Mackenbach commented:
#### Preview for one split
![74735334-c25da600-5250-11ea-8029-4ae1efa123c6](https://user-images.githubusercontent.com/17046398/75879305-75b2c700-5e1b-11ea-97de-5d3974994dea.png)

#### Preview for two splits
![74735341-c5f12d00-5250-11ea-9c19-cf2fd202d77c](https://user-images.githubusercontent.com/17046398/75879344-895e2d80-5e1b-11ea-8855-51943b65bc7c.png)

Rick-Mackenbach commented:
How can I fix the errors here?

jnothman commented:
The issue is in the call to set_points. There may be some incompatibility
between the version of matplotlib on that CI run and the version of numpy?
But double check the content and dtype what you pass to set_points.

Commit:
Merge remote-tracking branch 'upstream/master' into decision_threshold_plot

Commit:
Merge remote-tracking branch 'upstream/master' into decision_threshold_plot

Commit:
"Fix of set points error (required reshaping of array)"

Commit:
Merge remote-tracking branch 'upstream/master' into decision_threshold_plot

Commit:
"I added a few more lines of code, to do the preprocessing of text,
and setting of the bounding boxes in earlier code. Makes it easier for
interpretation"

Commit:
Merge remote-tracking branch 'upstream/master' into decision_threshold_plot

Commit:
"Included cov tests now"

Commit:
"Ignore unit test"

Commit:
Merge remote-tracking branch 'upstream/master' into decision_threshold_plot

jnothman reviewed (commented):
Hi @Rick-Mackenbach, you've added a lot of files here that shouldn't be. Please remove them from your branch, commit and push. Let us know if you need help doing this.

Commit:
Merge remote-tracking branch 'upstream/master' into decision_threshold_plot

Commit:
"Changes in test file, and made plotting of the first_decision_threshold
an argument within the plot_tree function."

Commit:
"Fix small linting error in the commented code"

Commit:
"Added the newline at the end of the code since this shows as a linting
error."

lucyleeow commented:
Hi @Rick-Mackenbach are you still working on this?
